src/libraries/Microsoft.Extensions.Hosting.Abstractions/src/HostingAbstractionsHostExtensions.cs

```diff
@@ -104,7 +104,7 @@ public static async Task WaitForShutdownAsync(this IHost host, CancellationToken
             await waitForStop.Task;
 
             // Host will use its default ShutdownTimeout if none is specified.
-            await host.StopAsync();
+            await host.StopAsync(token);
```

Comment:
Someone more familiar with these hosting abstractions should comment on whether this is correct.

Reply:
No, don't pass the token here. The token gets cancelled to unblock line 104. StopAsync has its own timeout logic.

Reply:
Thanks.  So this should be `default`.

Reply:
The `StopAsync` method has two overloads:

```cs
public interface IHost : IDisposable
{
     Task StopAsync(CancellationToken cancellationToken = default);
}

public static class HostingAbstractionsHostExtensions
{
    public static Task StopAsync(this IHost host, TimeSpan timeout)
    {
        return host.StopAsync(new CancellationTokenSource(timeout).Token);
    }
}
```
Two questions:
- Wouldn't `default` be ambiguous between `TimeSpan` and `CancellationToken`? Shouldn't I will explicitly pass `default(CancellationToken)`?
- When I opened the file with the extension overload, I got a warning saying that `new CancellationTokenSource(timeout)` is missing its `Dispose` call. I can get that fixed here, unless there are any objections or concerns.